include/Reader/reader.h

```diff
@@ -3279,6 +3292,102 @@ class ReaderBase {
 
   static bool rdrIsMethodVirtual(uint32_t MethodAttribs);
 
+  /// \brief Return MethodName for non helper and non native Methods.
+  ///
+  /// \param method  Method handle for the target Method.
+  /// \param classNamePtr out param, Module name.
+  /// \param JitInfo The jit interface for this method
+  /// \returns MethodName.
+  // TODO sandreenko : change cases to LLILC rules or transfer to coreclr.
+  const char *GetMethodName(CORINFO_METHOD_HANDLE method,
+                            const char **classNamePtr, ICorJitInfo *JitInfo);
+
+
+  /// \brief Return true if we process this Class with SIMD intrinsics.
+  ///
+  /// \param Class The class handle for the call target method's class.
+  /// \returns IRNode * LLVM::int(1 or 0).
+  virtual IRNode *generateIsHardwareAccelerated(CORINFO_CLASS_HANDLE Class) = 0;
+
+  /// \brief Return result of binary or unary operation on SIMD Vector Types.
+  /// \brief It gets arguments from stack.
+  ///
+  /// \param OperationCode code to be done. // TODO sandreenko : change to enum
+  /// later.
+  /// \param VectorSize Length of our Vector Type ( in floats).
+  /// \returns IRNode* !=0 -> Result of Intrinsic call,
+  ///  IRNode* == 0 -> we can't produce intrinsic.
```

Comment:
Instead of describing the return values as numbers, I'd recommend describing them abstractly:

```
\returns an \p IRNode representing the result of the intrinsic or nullptr if the intrinsic is not supported.
```